### PR TITLE
Cleanup Client.cs

### PR DESCRIPTION
**Pull request by OoLunar**
This PR:
- Removes several nullability warnings
- Sorts properties
- Adds documentation
- Removes extra whitespace
- Fixes incorrect usage of `Logger.Log`
- General logic cleanup

All changes have been tested and have no noticeable behavioral changes with commit c38662b4f35e979a4584a7184ab82cd1acc7a211

### CODE REVIEW

File: Obsidian/Client.cs
Comment (Craftplacer) [inline Obsidian/Client.cs:100]: Unnecessary comment, self-explanatory field name.

**Review by Craftplacer — APPROVED** [review]
Nice refactoring, I'm a bit unhappy about the excessive XMLdoc'ing on Client's fields, as they're most of the time very self-explanatory.

File: Obsidian/Client.cs
Comment (OoLunar) [reply Obsidian/Client.cs:100]: My apologies, that's what I get for mindlessly allowing Copilot to do it's own thing

**Comment by OoLunar** [comment]
> Nice refactoring, I'm a bit unhappy about the excessive XMLdoc'ing on Client's fields, as they're most of the time very self-explanatory.

I see two solutions: I could remove the self explanatory comments, which would make the documentation seem incomplete. Or I could give a more constructive explanation in the docs, much like I did with the brand property. Which would you prefer?

**Comment by Craftplacer** [comment]
> > Nice refactoring, I'm a bit unhappy about the excessive XMLdoc'ing on Client's fields, as they're most of the time very self-explanatory.
> 
> I see two solutions: I could remove the self explanatory comments, which would make the documentation seem incomplete. Or I could give a more constructive explanation in the docs, much like I did with the brand property. Which would you prefer?

Depends what other team members like @Naamloos or @Tides aim for.

File: Obsidian/Client.cs
Comment (Naamloos) [reply Obsidian/Client.cs:100]: You're using _copilot_ on our code base? 💀

Comment (OoLunar) [reply Obsidian/Client.cs:100]: Is there a stigma around copilot that I don't know about

File: Obsidian/Client.cs
Comment (Naamloos) [inline Obsidian/Client.cs:198]: Not sure what we decided on code standards, I'm not sure whether we require explicit variable typing..

**Review by Naamloos — APPROVED** [review]
Honestly LGTM, but I am not sure about changing explicit types to `var` for inline variables. Dunno what other maintainers / contribs think? I'd say fine, since I am not sure about my preferences yet. 

My job requires explicit typing in C# but iunno man this is not my job lmfao

**Comment by Naamloos** [comment]
^^ with that, I'll approve it for now. I'll leave it up to someone else to merge if they think it's fine :p

**Comment by Naamloos** [comment]
> > > Nice refactoring, I'm a bit unhappy about the excessive XMLdoc'ing on Client's fields, as they're most of the time very self-explanatory.
> > 
> > 
> > I see two solutions: I could remove the self explanatory comments, which would make the documentation seem incomplete. Or I could give a more constructive explanation in the docs, much like I did with the brand property. Which would you prefer?
> 
> Depends what other team members like @Naamloos or @Tides aim for.

I do personally really like to be explicit. Documentation is always a good thing. I don't think this is excessive at all.

File: Obsidian/Client.cs
Comment (Tides) [reply Obsidian/Client.cs:198]: I think its fine 🙏🏼

**Pull request merged**
